### PR TITLE
Adding support to send cognito user email notifications using custom domain

### PR DESCRIPTION
**Pull request by landonshumway-ia**
### Requirements List
- 

### Description List
- In order for the cognito user pool to send emails
from the custom domain, we need to set up an SES
identity along with necessary DNS records to verify
that the domain may be used to send emails.
- This also sets up an SNS topic which will forward
all bounce or complaint notifications to an
operations team email address.

### Testing List
- `yarn test:unit:all` should run without errors or warnings
- `yarn serve` should run without errors or warnings
- `yarn build` should run without errors or warnings
- Code review

Closes #149

### CODE REVIEW

File: backend/compact-connect/stacks/persistent_stack/user_email_notifications.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        Subscription(self, "FeedbackSubscription",
+                     topic=self.email_feedback_topic,
+                     protocol=SubscriptionProtocol.EMAIL,
+                     endpoint=operation_email)
+
+        self.config_set = ConfigurationSet(self, "ConfigSet")
+
+        self.config_set.add_event_destination(id="EmailFeedbackEventDestination",
+                                         destination=EventDestination.sns_topic(self.email_feedback_topic),
+                                         configuration_set_event_destination_name="EmailFeedbackDestination",
```
Comment (jusdino) [inline backend/compact-connect/stacks/persistent_stack/user_email_notifications.py:44]: Why force this resource name?

File: backend/compact-connect/stacks/persistent_stack/user_email_notifications.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        # grant SES the ability to encrypt bounce and complaint notifications using the KMS key
+        master_key.grant_encrypt_decrypt(ses_principal)
+
+
+        # Create SES Email Identity with DKIM enabled
+        self.email_identity = EmailIdentity(self, "EmailIdentity",
+            # by using the hosted zone resource, cdk will automatically
+            # create all the necessary DNS records for DKIM and SPF authentication
+            identity=Identity.public_hosted_zone(hosted_zone),
+            mail_from_domain=f"no-reply.{domain_name}",
```
Comment (jusdino) [inline backend/compact-connect/stacks/persistent_stack/user_email_notifications.py:61]: Can you talk me through the benefit of using this custom from domain?

**Review by jusdino — COMMENTED** [review]
Looks pretty good! I've got just a few questions to talk through when you have a sec.

File: backend/compact-connect/cdk.context.production-example.json [hunk 3]
```diff
@@ -8,6 +8,7 @@
         "region": "us-east-1",
         "connection_arn": "arn:aws:codestar-connections:us-east-1:000000000000:connection/11111111-1111-1111-111111111111",
         "notifications": {
+          "ses_feedback_forwarding_email": "justin@example.com",
```
[the remark this answers is not in the document]
Comment (landonshumway-ia) [reply backend/compact-connect/cdk.context.production-example.json:11]: I was attempting to fix the unit tests, but it turns out this line isn't needed so I'll remove it.

File: backend/compact-connect/cdk.context.production-example.json [hunk 4]
```diff
@@ -25,6 +26,7 @@
         "region": "us-east-1",
         "domain_name": "compactconnect.org",
         "notifications": {
+          "ses_feedback_forwarding_email": "justin@example.com",
```
[the remark this answers is not in the document]
Comment (landonshumway-ia) [reply backend/compact-connect/cdk.context.production-example.json:29]: Discussed f2f, we will want to separate concerns between email related notifications and application alerts.

File: backend/compact-connect/stacks/persistent_stack/user_email_notifications.py (hunk 1 above)
Comment (landonshumway-ia) [reply backend/compact-connect/stacks/persistent_stack/user_email_notifications.py:44]: ah, that's a copilot suggestion that we really don't need. I'll take that out.

File: backend/compact-connect/stacks/persistent_stack/user_email_notifications.py (hunk 2 above)
Comment (landonshumway-ia) [reply backend/compact-connect/stacks/persistent_stack/user_email_notifications.py:61]: discussed f2f, my understanding is that if you are going to use a sub domain for your from address (ie no-reply) this is needed for CDK to generate the needed TXT and MX records specifically to verify the use of that subdomain.

**Review by jusdino — APPROVED** [review]
Great!

**Pull request merged**